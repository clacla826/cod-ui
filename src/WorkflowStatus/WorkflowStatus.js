import React from 'react';
import PropTypes from 'prop-types';
import Grid from '@material-ui/core/Grid';
import ArrowLeftIcon from '@material-ui/icons/KeyboardArrowLeft';
import ArrowRightIcon from '@material-ui/icons/KeyboardArrowRight';
import WorkflowStatusItem from '../WorkflowStatusItem';
import useStyles from './styles';

/**
 * Displays the current state of an editorial workflow
 *
 * @example ./__docs__/WorkflowStatus.md
 */
const WorkflowStatus = ({ status }) => {
  const classes = useStyles();

  return (
    <Grid item className={classes.grow}>
      <Grid container justify="space-evenly" wrap="nowrap">
        <WorkflowStatusItem
          title="Declined"
          active={status === 'declined'}
          status="declined"
          caption="Proposal has been rejected by the parties."
        />
        <ArrowLeftIcon className={classes.arrow} />
        <WorkflowStatusItem
          title="Draft"
          active={status === 'draft'}
          status="draft"
          caption="Proposal is in writing stage, not yet submitted for an approval."
        />
        <ArrowRightIcon className={classes.arrow} />
        <WorkflowStatusItem
          title="Pending"
          active={status === 'pending'}
          status="pending"
          caption="Proposal is being vetted by the parties."
        />
        <ArrowRightIcon className={classes.arrow} />
        <WorkflowStatusItem
          title="Accepted"
          active={status === 'accepted'}
          status="accepted"
          caption="Proposal has been accepted and become an agreement."
        />
      </Grid>
    </Grid>
  );
};

WorkflowStatus.propTypes = {
  status: PropTypes.string.isRequired,
};

export default WorkflowStatus;
