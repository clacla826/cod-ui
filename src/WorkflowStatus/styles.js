import { makeStyles } from '@material-ui/styles';

const useStyles = makeStyles(() => ({
  arrow: {
    alignSelf: 'center',
    opacity: '.2',
  },
  grow: {
    flexGrow: 1,
  },
}));

export default useStyles;
