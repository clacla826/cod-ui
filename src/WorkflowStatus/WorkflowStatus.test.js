import React from 'react';
import { render } from '@testing-library/react';
import WorkflowStatus from './WorkflowStatus';

describe('WorkflowStatus', () => {
  it('matches the snapshot with label prop', () => {
    const { container } = render(<WorkflowStatus label="Agreement Status" status="draft" />);
    expect(container).toMatchSnapshot();
  });

  it('matches the snapshot without label prop', () => {
    const { container } = render(<WorkflowStatus status="draft" />);
    expect(container).toMatchSnapshot();
  });

  it('should not render with a label element', () => {
    const { queryByTestId } = render(<WorkflowStatus status="draft" />);
    expect(queryByTestId('workflow-label')).toBeNull();
  });
});
