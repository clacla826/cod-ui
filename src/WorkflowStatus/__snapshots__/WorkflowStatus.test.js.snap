// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkflowStatus matches the snapshot with label prop 1`] = `
<div>
  <div
    class="MuiGrid-root makeStyles-grow-2 MuiGrid-item"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-wrap-xs-nowrap MuiGrid-justify-xs-space-evenly"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation1 MuiCard-root makeStyles-activeCard-107 makeStyles-inactive-108 makeStyles-declinedStatus-109 MuiPaper-rounded"
      >
        <div
          class="MuiCardContent-root makeStyles-content-113"
        >
          <span
            class="MuiTypography-root makeStyles-inactive-108 makeStyles-declinedStatus-109 MuiTypography-overline"
          >
            Declined
          </span>
        </div>
        <div
          class="MuiCardContent-root makeStyles-content-113 makeStyles-caption-114"
        >
          <span
            class="MuiTypography-root MuiTypography-caption"
          >
            Proposal has been rejected by the parties.
          </span>
        </div>
      </div>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root makeStyles-arrow-1"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M15.41 16.59L10.83 12l4.58-4.59L14 6l-6 6 6 6 1.41-1.41z"
        />
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
      </svg>
      <div
        class="MuiPaper-root MuiPaper-elevation1 MuiCard-root makeStyles-activeCard-107 makeStyles-active-106 makeStyles-draftStatus-110 MuiPaper-rounded"
      >
        <div
          class="MuiCardContent-root makeStyles-content-113"
        >
          <span
            class="MuiTypography-root makeStyles-active-106 makeStyles-draftStatus-110 MuiTypography-overline"
          >
            Draft
          </span>
        </div>
        <div
          class="MuiCardContent-root makeStyles-content-113 makeStyles-caption-114"
        >
          <span
            class="MuiTypography-root MuiTypography-caption"
          >
            Proposal is in writing stage, not yet submitted for an approval.
          </span>
        </div>
      </div>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root makeStyles-arrow-1"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M8.59 16.59L13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"
        />
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
      </svg>
      <div
        class="MuiPaper-root MuiPaper-elevation1 MuiCard-root makeStyles-activeCard-107 makeStyles-inactive-108 makeStyles-pendingStatus-111 MuiPaper-rounded"
      >
        <div
          class="MuiCardContent-root makeStyles-content-113"
        >
          <span
            class="MuiTypography-root makeStyles-inactive-108 makeStyles-pendingStatus-111 MuiTypography-overline"
          >
            Pending
          </span>
        </div>
        <div
          class="MuiCardContent-root makeStyles-content-113 makeStyles-caption-114"
        >
          <span
            class="MuiTypography-root MuiTypography-caption"
          >
            Proposal is being vetted by the parties.
          </span>
        </div>
      </div>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root makeStyles-arrow-1"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M8.59 16.59L13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"
        />
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
      </svg>
      <div
        class="MuiPaper-root MuiPaper-elevation1 MuiCard-root makeStyles-activeCard-107 makeStyles-inactive-108 makeStyles-acceptedStatus-112 MuiPaper-rounded"
      >
        <div
          class="MuiCardContent-root makeStyles-content-113"
        >
          <span
            class="MuiTypography-root makeStyles-inactive-108 makeStyles-acceptedStatus-112 MuiTypography-overline"
          >
            Accepted
          </span>
        </div>
        <div
          class="MuiCardContent-root makeStyles-content-113 makeStyles-caption-114"
        >
          <span
            class="MuiTypography-root MuiTypography-caption"
          >
            Proposal has been accepted and become an agreement.
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`WorkflowStatus matches the snapshot without label prop 1`] = `
<div>
  <div
    class="MuiGrid-root makeStyles-grow-184 MuiGrid-item"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-wrap-xs-nowrap MuiGrid-justify-xs-space-evenly"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation1 MuiCard-root makeStyles-activeCard-289 makeStyles-inactive-290 makeStyles-declinedStatus-291 MuiPaper-rounded"
      >
        <div
          class="MuiCardContent-root makeStyles-content-295"
        >
          <span
            class="MuiTypography-root makeStyles-inactive-290 makeStyles-declinedStatus-291 MuiTypography-overline"
          >
            Declined
          </span>
        </div>
        <div
          class="MuiCardContent-root makeStyles-content-295 makeStyles-caption-296"
        >
          <span
            class="MuiTypography-root MuiTypography-caption"
          >
            Proposal has been rejected by the parties.
          </span>
        </div>
      </div>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root makeStyles-arrow-183"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M15.41 16.59L10.83 12l4.58-4.59L14 6l-6 6 6 6 1.41-1.41z"
        />
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
      </svg>
      <div
        class="MuiPaper-root MuiPaper-elevation1 MuiCard-root makeStyles-activeCard-289 makeStyles-active-288 makeStyles-draftStatus-292 MuiPaper-rounded"
      >
        <div
          class="MuiCardContent-root makeStyles-content-295"
        >
          <span
            class="MuiTypography-root makeStyles-active-288 makeStyles-draftStatus-292 MuiTypography-overline"
          >
            Draft
          </span>
        </div>
        <div
          class="MuiCardContent-root makeStyles-content-295 makeStyles-caption-296"
        >
          <span
            class="MuiTypography-root MuiTypography-caption"
          >
            Proposal is in writing stage, not yet submitted for an approval.
          </span>
        </div>
      </div>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root makeStyles-arrow-183"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M8.59 16.59L13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"
        />
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
      </svg>
      <div
        class="MuiPaper-root MuiPaper-elevation1 MuiCard-root makeStyles-activeCard-289 makeStyles-inactive-290 makeStyles-pendingStatus-293 MuiPaper-rounded"
      >
        <div
          class="MuiCardContent-root makeStyles-content-295"
        >
          <span
            class="MuiTypography-root makeStyles-inactive-290 makeStyles-pendingStatus-293 MuiTypography-overline"
          >
            Pending
          </span>
        </div>
        <div
          class="MuiCardContent-root makeStyles-content-295 makeStyles-caption-296"
        >
          <span
            class="MuiTypography-root MuiTypography-caption"
          >
            Proposal is being vetted by the parties.
          </span>
        </div>
      </div>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root makeStyles-arrow-183"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M8.59 16.59L13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"
        />
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
      </svg>
      <div
        class="MuiPaper-root MuiPaper-elevation1 MuiCard-root makeStyles-activeCard-289 makeStyles-inactive-290 makeStyles-acceptedStatus-294 MuiPaper-rounded"
      >
        <div
          class="MuiCardContent-root makeStyles-content-295"
        >
          <span
            class="MuiTypography-root makeStyles-inactive-290 makeStyles-acceptedStatus-294 MuiTypography-overline"
          >
            Accepted
          </span>
        </div>
        <div
          class="MuiCardContent-root makeStyles-content-295 makeStyles-caption-296"
        >
          <span
            class="MuiTypography-root MuiTypography-caption"
          >
            Proposal has been accepted and become an agreement.
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
